lib/containers/runtime.pm

```diff
@@ -131,18 +214,85 @@ sub remove_image {
     $self->_rt_assert_script_run("rmi -f $image_name");
 }
 
+=head2 remove_container
+
+Remove a container from the pool.
+
+=cut
+sub remove_container {
+    my ($self, $container_name) = @_;
+    $self->_rt_assert_script_run("rm -f $container_name");
+}
+
+=head2 confirm_image_in_host_registry
+
+Returns true if host contains C<img> or false.
+
+=cut
+sub confirm_image_in_host_registry {
+    my ($self, $img) = shift;
+    my @lregistry = $self->enum_images();
+    $img =~ @lregistry;
+}
+
+=head2 setup_registry
+
+Updates the registry files for the running container runtime to allow access to
+insecure registries.
+
+=cut
+sub setup_registry {
+    my ($self) = shift;
+    my $registry = registry_url();
+    if ($self->runtime =~ /docker/) {
+        # Allow our internal 'insecure' registry
+        assert_script_run(
+'echo "{ \"debug\": true, \"insecure-registries\" : [\"localhost:5000\", \"registry.suse.de\", \"' . $registry . '\"] }" > /etc/docker/daemon.json');
+        assert_script_run('cat /etc/docker/daemon.json');
+        systemctl('restart docker');
+    } elsif ($self->runtime =~ /podman/) {
+        assert_script_run "curl " . data_url('containers/registries.conf') . " -o /etc/containers/registries.conf";
+        assert_script_run "chmod 644 /etc/containers/registries.conf";
+        file_content_replace("/etc/containers/registries.conf", REGISTRY => $registry);
+    } else {
+        die "Unsupported runtime - " . $self->runtime;
+    }
+}
+
 =head2 cleanup_system_host
 
-Remove containers and then all the images respectively and then make sure that everything was cleaned up.
+Remove containers and then all the images respectively and then make sure that everything was cleaned up
+by default, unless C<confirm> is defined to 0 exclusively.
```

Comment:
This comment is confusing and I don't understand what `confirm = 0` should do.

I'd rename `confirm` to `assert` within the function and would change the comment to something like

    Remove containers and then all the images respectively. Asserts that everything was cleaned up unless c<assert> is set to 0.

Reply:
Done